packages/cli/ts/commands/deployPoll.ts

```diff
@@ -49,6 +50,18 @@ export const deployPoll = async ({
 
   const maci = maciAddress || maciContractAddress;
 
+  const maciContract = MACIFactory.connect(maci, signer);
+  const pollId = await maciContract.nextPollId();
+
+  // check if we have a signupGatekeeper already deployed or passed as arg
+  let signupGatekeeperContractAddress =
+    gatekeeperAddress || (await readContractAddress(`SignUpGatekeeper-${pollId.toString()}`, network?.name));
+
+  if (!signupGatekeeperContractAddress) {
+    const contract = await deployFreeForAllSignUpGatekeeper(signer, true);
```

Comment:
I don't quite understand this part: why we deploy `FreeForAllSignupGatekeeper` contract instead of letting them indicate what kind of gatekeeper they wanna deploy?

Reply:
this is the same logic as in maci deployment in the cli/sdk. You are supposed to deploy it separately unless you just want the free for all